### PR TITLE
Build Test Akka Cluster Metrics with sigar library.

### PR DESCRIPTION
**Pull request by RayRoestenburg**

### CODE REVIEW

File: .sbtopts-nightly-sigar [hunk 1]
```diff
@@ -0,0 +1,19 @@
+-J-XX:+UseG1GC -Xms3g -Xmx3g
```
Comment (andreaTP) [inline .sbtopts-nightly-sigar:1]: let's keep the jvm settings in one place https://github.com/akka/akka/blob/master/.jvmopts-ci
and is fine to remove:
```
-J-XX:+PrintGCDetails 
-J-XX:+PrintGCTimeStamps 
```

File: .github/workflows/nightly-sigar.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+      - name: Set up JDK 8
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.8-0
+      - name: Test
+        run: |-
+          cp .sbtopts-nightly-sigar .sbtopts
+          sbt clean
```
Comment (andreaTP) [inline .github/workflows/nightly-sigar.yml:23]: one single sbt command?

File: .github/workflows/nightly-sigar.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: Akka Nightly Sigar Build
```
Comment (andreaTP) [inline .github/workflows/nightly-sigar.yml:1]: I'm confused when I see Sigar and Cluster Metrics, I think we can name everything "Cluster Metrics", wdyt?

**Review by andreaTP — COMMENTED** [review]
minor comments

File: .github/workflows/nightly-sigar.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+          fetch-depth: 0
+      - name: Set up JDK 8
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.8-0
+      - name: Test
+        run: |-
+          cp .sbtopts-nightly-sigar .sbtopts
+          sbt clean 
+          sbt akka-cluster-metrics/test
```
Comment (raboof) [inline .github/workflows/nightly-sigar.yml:25]: Aren't these ran as part of the 'regular' nightly tests? What is the motivation for running them separately? Can we add that as a comment?

(I understand you might not know, but I don't know either, might be good to understand before we add this job?)

File: .sbtopts-nightly-sigar [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+-J-XX:-UseBiasedLocking 
+-J-XX:+UseCompressedOops 
+-J-XX:+PrintGCDetails 
+-J-XX:+PrintGCTimeStamps 
+-J-XX:MaxPermSize=512M 
+-J-XX:MetaspaceSize=512M 
+-J-Xss2m 
+-J-XX:ReservedCodeCacheSize=256m 
+-Djava.security.egd=file:/dev/./urandom
+-Dakka.test.sigar=true
```
Comment (raboof) [inline .sbtopts-nightly-sigar:11]: Ah, perhaps the regular tests are ran without this option?

File: .github/workflows/nightly-sigar.yml (hunk 3 above)
Comment (raboof) [reply .github/workflows/nightly-sigar.yml:1]: I guess the difference is that this job runs the cluster metrics tests with 'sigar' enabled, and the regular build runs them without that library?

Comment (RayRoestenburg) [reply .github/workflows/nightly-sigar.yml:1]: That is my guess too.. let me rename.

File: .github/workflows/nightly-sigar.yml (hunk 4 above)
Comment (RayRoestenburg) [reply .github/workflows/nightly-sigar.yml:25]: This is with the sigar library enabled though.

File: .sbtopts-nightly-sigar (hunk 5 above)
Comment (RayRoestenburg) [reply .sbtopts-nightly-sigar:13]: yes.

Comment (RayRoestenburg) [reply .sbtopts-nightly-sigar:13]: I have no idea how relevant the specific test for sigar is TBH.

File: .sbtopts-nightly-sigar (hunk 1 above)
Comment (RayRoestenburg) [reply .sbtopts-nightly-sigar]: They are different so will add a `.jvmopts-nightly-cluster-metrics-sigar` and copy it.

Comment (ignasi35) [reply .sbtopts-nightly-sigar]: Each single job has a completely different set of arguments.

**Review by ignasi35 — APPROVED** [review]

File: .sbtopts-nightly-sigar (hunk 1 above)
Comment (raboof) [reply .sbtopts-nightly-sigar]: let's unify them as much as possible

**Comment by raboof** [comment]
I suspect we'll see a lot of flaky test failures, also in these PR's.

I know it's hard, but let's make sure we create issues for those failures, keep track of which ones are most common/important, and fix some.

**Review by raboof — COMMENTED** [review]
How do we get notified of failures?

I don't have a strong preference between emailing and creating issues, but I guess we should set up something?

**Comment by ignasi35** [comment]
> I don't have a strong preference between emailing and creating issues, but I guess we should set up something?

Since GHActiosn will be veeery flaky on the first few iterations I'd stick to e-mail. It'll be less noisy and easier to cleanup as we get the gist of it and keep things under control.

File: .github/workflows/nightly-cluster-metrics-sigar.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+      - name: Set up JDK 8
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.8.0
+      - name: Test
+        run: |-
+          cp .sbtopts-nightly-cluster-metrics-sigar .sbtopts
+          sbt -jvm-opts .jvmopts-nightly-cluster-metrics-sigar clean akka-cluster-metrics/test
```
Comment (ignasi35) [inline .github/workflows/nightly-cluster-metrics-sigar.yml:24]: Setup e-mail:

https://github.com/akka/akka/pull/30449/files#diff-f830b2d6328b69875d6e2629be69a5d45e3c66459ed82943dbeafab88fc6c284R30-R45

**Review by ignasi35 — APPROVED** [review]

File: .jvmopts-nightly-cluster-metrics-sigar [hunk 7]
```diff
@@ -0,0 +1,11 @@
+-XX:+UseG1GC
```
Comment (andreaTP) [inline .jvmopts-nightly-cluster-metrics-sigar:1]: all those settings can be in the CI default `.jvmopts-ci` IMHO

**Review by raboof — APPROVED** [review]
ofc more things can be improved, but this LGTM as a start

**Review by raboof — COMMENTED** [review]
Failure notification still needs to be added though

File: .sbtopts-nightly-sigar (hunk 1 above)
Comment (RayRoestenburg) [reply .sbtopts-nightly-sigar]: I'll try to run the standard `.jvmopts-ci`

**Review by raboof — APPROVED** [review]

File: .github/workflows/nightly-sigar.yml (hunk 2 above)
Comment (RayRoestenburg) [reply .github/workflows/nightly-sigar.yml]: change to how other builds are done now, with sbt options in-line

File: .sbtopts-nightly-sigar [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+-J-XX:MetaspaceSize=512M 
+-J-Xss2m 
+-J-XX:ReservedCodeCacheSize=256m 
+-Djava.security.egd=file:/dev/./urandom
+-Dakka.test.sigar=true 
+-Dakka.cluster.assert=on  
+-Dakka.test.metrics.reporters.0=console 
+-Dakka.test.metrics.reporters.1=graphit 
+-Dakka.test.metrics.reporter.graphite.prefix=jenkins 
+-Dakka.test.metrics.reporter.graphite.host=54.72.154.120
```
[the remark this answers is not in the document]
Comment (RayRoestenburg) [reply .sbtopts-nightly-sigar]: removed.

File: .sbtopts-nightly-sigar [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+-J-XX:ReservedCodeCacheSize=256m 
+-Djava.security.egd=file:/dev/./urandom
+-Dakka.test.sigar=true 
+-Dakka.cluster.assert=on  
+-Dakka.test.metrics.reporters.0=console 
+-Dakka.test.metrics.reporters.1=graphit 
+-Dakka.test.metrics.reporter.graphite.prefix=jenkins 
+-Dakka.test.metrics.reporter.graphite.host=54.72.154.120 
+-Dakka.sbt.statsd=false 
+-Dakka.test.multi-node.java=/usr/lib/jvm/jdk8u192-b12/bin/java
```
[the remark this answers is not in the document]
Comment (RayRoestenburg) [reply .sbtopts-nightly-sigar]: removed.

File: .jvmopts-nightly-cluster-metrics-sigar (hunk 7 above)
Comment (RayRoestenburg) [reply .jvmopts-nightly-cluster-metrics-sigar]: not sure if we really need G1 GC, will try without.

**Comment by RayRoestenburg** [comment]
@raboof added email notification on failure.

**Comment by RayRoestenburg** [comment]
Build Scala 3 failed on flaky BalancingSpec. Nightly Akka Cluster Metrics Test with Sigar / Test Akka Cluster Metrics succeeded. Ready for another review @andreaTP @ignasi35 @raboof

**Comment by raboof** [comment]
Issue for flakey BalancingSpec is #30359

**Review by raboof — APPROVED** [review]

**Pull request merged**